pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -528,18 +541,21 @@ func (e *eksOperatorController) deployEKSOperator() error {
 			},
 		}
 
+		desiredApp.Spec.ValuesYaml = valuesYaml
+
 		// k3s upgrader doesn't exist yet, so it will need to be created
 		if _, err = e.appClient.Create(desiredApp); err != nil {
 			return err
 		}
 	} else {
-		if app.Spec.ExternalID == latestVersionID {
+		if app.Spec.ExternalID == latestVersionID && app.Spec.ValuesYaml == valuesYaml {
 			// app is up to date, no action needed
 			return nil
 		}
 		logrus.Info("updating EKS operator in local cluster's system project")
 		desiredApp := app.DeepCopy()
 		desiredApp.Spec.ExternalID = latestVersionID
+		desiredApp.Spec.ValuesYaml = valuesYaml
```

Comment:
So based on how the valuesYaml works, this should get overlaid on top of the existing values.yaml. And this would only happen if the user didn't set environment variables for it?

Reply:
Oop! I did read the env parsing incorrectly so you can ignore that second half of my statement

Reply:
Regarding the first part, that is correct.